config/internal/configsource/README.md

```diff
@@ -0,0 +1,101 @@
+                          |                                                     |                             
+                          | logica  YAML config:                                |                             
+                          | +-------------------------------------------------+ |                             
+                          | |receivers:                                       | |                             
+                          | |  zipkin:                                        | |                             
+                          | |  jaeger:                                        | |                             
+                          | |  otlp:                                          | |                             
+                          | |exporters:                                       | |                             
+                          | |  zipkin:                                        | |                             
+                          | |  jaeger:                                        | |                             
+                          | |  otlp:                                          | |                             
+                          | |service:                                         | |                             
+                          | |  pipelines:                                     | |                             
+                          | |    trace:                                       | |                             
+                          | |      receivers: [zipkin, jaeger, otlp]          | |                             
+                          | |      exporters: [otlp]                          | |                             
+                          | +-------------------------------------------------+ |                             
+                          +-----------------------------------------------------+                             
+                                                                                                              
+                                                                                                                                                                                                                            
+```
+
+## Watching for Updates
+After the configuration was processed the `configsource.Manager` can be used as a single point to watch for updates in
+the configuration data retrieved via the config sources used to process the “initial” configuration and to generate
+the“effective” one.
+
+The `configsource.Manager` does that by wrapping calls to the `WatchForUpdate()` method of each `configsource.Retreived`
+object that was used during the configuration processing. It also controls the lifecycle of all `configsource.Session`
+objects created to get the `configsource.Retrieved` objects. 
```

Comment:
Maybe expand a bit more on what is `configsource.Retreived` and what is `configsource.Session`. Neither of these are visible on the diagram. Is it possible to add them or have a different diagram that shows them?

Reply:
Will do.

Reply:
Done. Remember that there is also some info about [configsorce.Retrieved](https://github.com/open-telemetry/opentelemetry-collector/blob/b336d3df8b9d483c1e07e4bc216d788289da7e9d/config/experimental/configsource/component.go#L83) and [configsource.Session](https://github.com/open-telemetry/opentelemetry-collector/blob/b336d3df8b9d483c1e07e4bc216d788289da7e9d/config/experimental/configsource/component.go#L57) at the `config/experimental/configsource` package. We should probably re-organize these packages, but after we get the manager code updated.